File: wibb-android/app/src/main/java/com/example/wibb/data/Store.java
package com.example.wibb.data;

public class Store implements GridDisplayable {

    private String name;
    private int icon;

    public Store(String name, int icon) {
        this.name = name;
        this.icon = icon;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public int getIcon() {
        return icon;
    }

    public void setIcon(int icon) {
        this.icon = icon;
    }

    @Override
    public int getDrawable() {
        return getIcon();
    }

    @Override
    public String getText() {
        return getName();
    }
}

File: wibb-android/app/src/main/java/com/example/wibb/data/Store.kt
package com.example.wibb.data

import org.json.JSONObject

class Store(var name: String, var icon: String): GridDisplayable {

    override val iconurl: String
        get() = icon

    override val text: String
        get() = name

    companion object {
        fun fromJSON(jstore: JSONObject): Store {
            return Store(
                jstore.getString("name"),
                jstore.getString("icon")
            )
        }
    }

    fun toJSON(): JSONObject{
        val jo = JSONObject()
        jo.put("name", this.name)
        jo.put("icon", this.icon)
        return jo
    }
}
